src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Responses.cs

```diff
@@ -441,4 +444,290 @@ public async Task RequestDelegateWritesComplexStructReturnValueAsJsonResponseBod
+    {
+        get
+        {
+            var testAction = """
+app.MapPost("/", () => "String Test");
+""";
+
+            var taskTestAction = """
+app.MapPost("/", () => Task.FromResult("String Test"));
+""";
+
+            var valueTaskTestAction = """
+app.MapPost("/", () => ValueTask.FromResult("String Test"));
+""";
+
+            var staticTestAction = """
+app.MapPost("/", StaticTestAction);
+static string StaticTestAction() => "String Test";
+""";
+
+            var staticTaskTestAction = """
+app.MapPost("/", StaticTaskTestAction);
+static Task<string> StaticTaskTestAction() => Task.FromResult("String Test");
+""";
+
+            var staticValueTaskTestAction = """
+app.MapPost("/", StaticValueTaskTestAction);
+static ValueTask<string> StaticValueTaskTestAction() => ValueTask.FromResult("String Test");
+""";
+
+            var staticStringAsObjectTestAction = """
+app.MapPost("/", StaticTestAction);
+static object StaticTestAction() => "String Test";
+""";
+
+            var staticStringAsTaskObjectTestAction = """
+app.MapPost("/", StaticTaskTestAction);
+static Task<object> StaticTaskTestAction() => Task.FromResult<object>("String Test");
+""";
+
+            var staticStringAsValueTaskObjectTestAction = """
+app.MapPost("/", StaticValueTaskTestAction);
+static ValueTask<object> StaticValueTaskTestAction() => ValueTask.FromResult<object>("String Test");
+""";
+
+            return new List<object[]>
+                {
+                    new object[] { testAction },
+                    new object[] { taskTestAction },
+                    new object[] { valueTaskTestAction },
+                    new object[] { staticTestAction },
+                    new object[] { staticTaskTestAction },
+                    new object[] { staticValueTaskTestAction },
+
+                    new object[] { staticStringAsObjectTestAction },
+
+                    new object[] { staticStringAsTaskObjectTestAction },
+                    new object[] { staticStringAsValueTaskObjectTestAction },
+
+                };
+        }
+    }
+
+    [Theory]
+    [MemberData(nameof(StringResult))]
+    public async Task RequestDelegateWritesStringReturnValueAndSetContentTypeWhenNull(string source)
+    {
+        var (_, compilation) = await RunGeneratorAsync(source);
+        var endpoint = GetEndpointFromCompilation(compilation);
+
+        var httpContext = CreateHttpContext();
+        httpContext.RequestServices = new ServiceCollection()
+            .AddSingleton(LoggerFactory)
+            .AddSingleton(Options.Create(new JsonOptions()))
+            .BuildServiceProvider();
+
+        await endpoint.RequestDelegate(httpContext);
+
+        await VerifyResponseBodyAsync(httpContext, "String Test");
+        Assert.Equal("text/plain; charset=utf-8", httpContext.Response.ContentType);
+    }
+
+    //[Theory]
+    //[MemberData(nameof(StringResult))]
+    //public async Task RequestDelegateWritesStringReturnDoNotChangeContentType(Delegate @delegate)
```

Comment:
?

Reply:
Ah missed this when i was going through tidying up for PR submission. Thanks!